src/components/network-chart/index.js

```diff
@@ -60,74 +63,127 @@ class NetworkChart extends React.PureComponent<Props> {
+
+    let markerLength = markerDuration / timeRangeTotal * 75;
+
+    if (markerLength < 0.15) {
+      markerLength = 0.15;
+    }
+
+    return markerLength;
+  }
+
+  _getMarkerStyling(marker) {
+    const markerLength = this._getMarkerLength(marker.dur);
+    const markerPosition = this._getMarkerPosition(marker.startTime);
+
+    const markerStyling = {
+      width: markerLength + '%',
+      left: markerPosition + '%',
+    };
+    return markerStyling;
+  }
+
+  // Create row with correct details
+  _renderRow(nodeId: any, index: number) {
+    const markersPerRange = this._filterMarkersPerRange(
+      this.props.timeRange,
+      this.props.markers
+    );
+
+    const itemClassName = (
+      'item ' + markersPerRange[index].status
+    ).toLowerCase();
+
+    return (
+      <section className={itemClassName}>
+        <div className="status" />
+        <div className="itemLabel">
+          {markersPerRange[index].URI} - {markersPerRange[index].dur}
+        </div>
+        <div
+          className="itemBar"
+          style={this._getMarkerStyling(markersPerRange[index])}
+        >
+          &nbsp;
+        </div>
+      </section>
+    );
+  }
+
+  _filterMarkersPerRange(timeRange: Object, markers: []) {
+    return markers.filter(function(marker) {
+      const updateRange =
+        timeRange.start < marker.startTime && timeRange.end > marker.endTime;
+      return updateRange;
+    });
+  }
+
   render() {
-    const {
-      maxNetworkRows,
-      timeRange,
-      threadIndex,
-      networkTimingRows,
-      markers,
-      previewSelection,
-      updatePreviewSelection,
-    } = this.props;
+    const { timeRange, markers } = this.props;
 
-    if (!networkTimingRows.length) {
+    //eslint-disable-next-line
+    const networkMarkersPerRange = this._filterMarkersPerRange( timeRange, markers );
+    if (!networkMarkersPerRange.length) {
       return <NetworkChartEmptyReasons />;
     }
 
     // The viewport needs to know about the height of what it's drawing, calculate
     // that here at the top level component.
-    const maxViewportHeight = maxNetworkRows * ROW_HEIGHT;
+    // const maxViewportHeight = maxNetworkRows * ROW_HEIGHT;
 
     return (
       <div className="networkChart">
-        <MarkerChartCanvas
-          key={threadIndex}
-          viewportProps={{
-            timeRange,
-            previewSelection,
-            maxViewportHeight,
-            viewportNeedsUpdate,
-            maximumZoom: this.getMaximumZoom(),
-          }}
-          chartProps={{
-            networkTimingRows,
-            markers,
-            updatePreviewSelection,
-            rangeStart: timeRange.start,
-            rangeEnd: timeRange.end,
-            rowHeight: ROW_HEIGHT,
-            threadIndex,
-          }}
+        <VirtualList
+          className="treeViewBody"
+          items={networkMarkersPerRange}
+          renderItem={this._renderRow}
+          itemHeight={ROW_HEIGHT}
+          columnCount={1}
```

Comment:
Those fixed values will be updated. @julienw some pointers here would be great on which ones we need for this component and which not. thanks!

Reply:
The VirtualList could use some documentation on its API :)

I'll do a rough explanations on each element:
* className: self-explanatory
* items: self-explanatory: the items to be displayed will be taken out of this `items` array depending on the scroll position
* renderItem: the function that will render 1 item
* itemHeight: the height for 1 item: this is necessary for the component to calculate which items to display. This can depend on the component of course
* columnCount: the virtual list can display several columns; which column to render is passed to the `renderItem` callback
* focusable: controls the tabIndex attribute
* onKeyDown: makes it possible to handle keyboard input
* specialItems: forces redrawing one item -- useful to handle selection
* disableOverscan: controls the overscan, which is the fact that we render elements ahead or above the scroll position
* onCopy: controls what to do on a keyboard copy action (honestly I'm not sure why this is in VirtualList)
* getHoveredItemInfo: I don't see this in the current VirtualList ?
* containerWidth: the TreeView can expand to the right a lot but I suspect it's not the case for the network chart. Maybe we can make it optional and make the layout adjust automatically.

Reply:
Cool, thanks for the info!